### PR TITLE
Floor follow

### PR DESCRIPTION
**Pull request by mjstn**
This is the first phase of changes to follow.py abilities.   This set of changes allows default follow_me operation with fiducial 49 but then adds the ability of the follow node to act as a server.
New messages FollowerCommand.msg is sent on the inbound topic and status sent back out on FollowerStatus.msg messages.

The readme explains this new mode.

It is best to put this mode into master at this time because Phase II will be a significant new addition and we cannot wait forever as what this pull request does is a new ability usable by itself for users to write their own clients starting with the example,  follower_client.py in this set of changes.

### CODE REVIEW

**Comment by mjstn** [comment]
I will do a squash and merge at the time of the merge.
Also, I have been sitting on this for a VERY LONG time so it is well overdue.

**Comment by mjstn** [comment]
I am adding Jim in the hopes that somebody will unblock this pull request.  It is more in his area anyway.  Thanks

File: custom_messages/msg/FollowerCommand.msg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+string  commandType
```
Comment (rohbotics) [inline custom_messages/msg/FollowerCommand.msg]: Instead of Strings these should be based on integers with defined constant values. This way there is less coordination required between nodes, and the message definition itself contains a lot of the information needed.

Example is the `radiation_type` here http://docs.ros.org/melodic/api/sensor_msgs/html/msg/Range.html

Comment (rohbotics) [inline custom_messages/msg/FollowerCommand.msg:1]: Instead of Strings these should be based on integers with defined constant values. This way there is less coordination required between nodes, and the message definition itself contains a lot of the information needed.

Example is the `radiation_type` here http://docs.ros.org/melodic/api/sensor_msgs/html/msg/Range.html

File: custom_messages/aruco_detect.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+aruco_detect:
```
Comment (rohbotics) [inline custom_messages/aruco_detect.yaml:1]: Why is this in the message package?

File: custom_messages/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+
+
+# Nodes
```
Comment (rohbotics) [inline custom_messages/README.md:3]: This readme seems unrelated to this package.

**Review by rohbotics — CHANGES_REQUESTED** [review]
Hi Mark, 
Sorry it took so long for the review.

I am confused on the overall architecture here, it seems terribly complicated for something that could just be a standalone node. Can we get on the phone to discuss this?

Why is there a new package called "custom_messages"? Can this be integrated with the follow package? I don't see a circular dependency issue. If we need to keep it, we will need a better name "custom" is not descriptive, and ROS uses "msgs" not "messages" in package names.

custom_messages/nodes/fiducial_tracker.pyc Should not be checked in, it is an autogenerated python bytecode file. 

I question the need to modify the existing follow node instead of creating a new floor following package. The original follower code was simple-enough example that showed how to command and control Magni with `cmd_vel`.

**Comment by mjstn** [comment]
Standing down from this approach in favor of Actionlib server in floor_follow folder.
